webpack/move_to_foreman/components/common/table/index.js

```diff
@@ -32,27 +33,62 @@ export const TableBody = (props) => {
   )
 }
 
-class Table extends React.Component {
+export class Table extends React.Component {
   isEmpty() {
     return this.props.rows.length === 0 && this.props.bodyMessage === undefined;
   }
 
   render() {
-    const { columns, rows, emptyState, bodyMessage } = this.props;
-    return this.isEmpty() ? (
-      <EmptyState {...emptyState} />
-    ) : (
-      <PfTable.PfProvider
-        className="table-fixed"
-        striped
-        bordered
-        hover
-        columns={columns}
-      >
-        <PfTable.Header />
-        <TableBody columns={columns} rows={rows} message={bodyMessage} />
-      </PfTable.PfProvider>
-    );
+    const { columns, rows, emptyState, bodyMessage, children, itemCount, pagination, onPaginationChange, ...otherProps } = this.props;
+
+    let paginationComponent;
+    if (itemCount && pagination) {
+      paginationComponent = (
+        <PaginationRow
+          viewType="table"
+          itemCount={itemCount}
+          pagination={pagination}
+          onChange={onPaginationChange}
+        />
+      );
+    }
+
+    if (this.isEmpty()) {
+      return (<EmptyState {...emptyState} />);
+    } else if ( children !== undefined ) {
+      return (
+        <div>
+          <PfTable.PfProvider
+            className="table-fixed"
+            striped
+            bordered
+            hover
+            columns={columns}
+            {...otherProps}
+          >
+            {children}
+          </PfTable.PfProvider>
+          {paginationComponent}
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <PfTable.PfProvider
+            className="table-fixed"
+            striped
+            bordered
+            hover
+            columns={columns}
+            {...otherProps}
+          >
```

Comment:
Is there a way to not repeat `PfTable.PfProvider` in this if/else? Asking as I will have to add to it and doing so will duplicate more code